File: 2026-04-April-GFG-POTD/April 26 - Common in 3 Sorted Arrays/Solution.py
class Solution:
    def commonElements(self, a, b, c):
        res = []
        
        i = j = k = 0
        
        while i < len(a) and j < len(b) and k < len(c):
            
            # If all three are equal
            if a[i] == b[j] == c[k]:
                # Avoid duplicates
                if not res or res[-1] != a[i]:
                    res.append(a[i])
                i += 1
                j += 1
                k += 1
            
            elif a[i] < b[j]:
                i += 1
            
            elif b[j] < c[k]:
                j += 1
            
            else:
                k += 1
        
        return res

File: 2026-04-April-GFG-POTD/April 26 - Common in 3 Sorted Arrays/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> commonElements(vector<int> &a, vector<int> &b, vector<int> &c) {
        vector<int> res;
        
        int i = 0, j = 0, k = 0;
        
        while (i < a.size() && j < b.size() && k < c.size()) {
            
            // If all three are equal
            if (a[i] == b[j] && b[j] == c[k]) {
                // Avoid duplicates
                if (res.empty() || res.back() != a[i]) {
                    res.push_back(a[i]);
                }
                i++; j++; k++;
            }
            else if (a[i] < b[j]) {
                i++;
            }
            else if (b[j] < c[k]) {
                j++;
            }
            else {
                k++;
            }
        }
        
        return res;
    }
};
